test/integration/targets/iam_user_info/tasks/main.yml

```diff
@@ -0,0 +1,105 @@
+---
+- name: set up aws connection info
+  module_defaults:
+    group/aws:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token | default(omit) }}"
+      region: "{{ aws_region }}"
+  block:
+  - name: ensure ansible user exists
+    iam_user:
+      name: '{{ test_user }}'
+      state: present
+
+  - name: get info on IAM user(s) on path
+    iam_user_info:
+      path: '{{ test_path }}'
+      name: '{{ test_user }}'
+    register: iam_user_info
+
+  - name: assert correct number of users on path are returned
+    assert:
+      that:
+        - iam_user_info.iam_users | length == 1
+
+  - name: assert correct keys for path are returned
+    assert:
+      that:
+        - iam_user_info.iam_users[0].arn is not none
```

Comment:
niggle: You can do a quick test that this is actually an ARN:
```
....arn.startswith("arn:aws:iam")
....arn.endswith("user/{{ test_user }}")
```

Reply:
https://github.com/ansible/ansible/pull/23382/commits/9f67467fdd9c5ab480bc1d8e0d2aa20f20599b7d adds some more detailed checks.